google-ads/src/main/java/com/google/ads/googleads/lib/GoogleAdsClient.java

```diff
@@ -551,10 +551,8 @@ public GoogleAdsClient build() {
       // Provides the credentials to the primer to preemptively get these ready for usage.
       Primer.getInstance().ifPresent(p -> p.primeCredentialsAsync(getCredentials()));
       // Proceeds with creating the client library instance.
-      TransportChannelProvider transportChannelProvider = getTransportChannelProvider();
-      if (transportChannelProvider.needsHeaders()) {
-        transportChannelProvider = transportChannelProvider.withHeaders(getHeaders());
-      }
+      TransportChannelProvider transportChannelProvider =
```

Comment:
The logging interceptor also needs to be updated.

Notice in `newBuilder()` that it's initialized with a copy of the headers. This means that we'll need to recreate it every time `build()` is called.

https://screenshot.googleplex.com/5uBDrqtSsG4X3jo

Additionally we should preserve any additional interceptors added by clients via `GoogleAdsClient.Builder.setTransportChannelProvider()`.

Reply:
Open comment here